www/apps/book/app/advanced-development/modules/query/page.mdx

```diff
@@ -0,0 +1,225 @@
+
+In your resources, such as API routes or workflows, you can resolve the Query to fetch data across custom modules and Medusa’s commerce modules.
+
+---
+
+## Query Example
+
+For example, create the route `src/api/store/query/route.ts` with the following content:
+
+export const exampleHighlights = [
+  ["13", "", "Resolve the Query from the Medusa container."],
+  ["15", "graph", "Run a query to retrieve data."],
+  ["16", "entryPoint", "The name of the data model you're querying."],
+  ["17", "fields", "An array of the data model’s properties to retrieve in the result."],
+]
+
+```ts title="src/api/store/query/route.ts" highlights={exampleHighlights} apiTesting testApiMethod="GET" testApiUrl="http://localhost:9000/store/query" collapsibleLines="1-8" expandButtonLabel="Show Imports"
+import {
+  MedusaRequest,
+  MedusaResponse,
+} from "@medusajs/medusa"
+import {
+  ContainerRegistrationKeys,
+} from "@medusajs/utils"
+
+export const GET = async (
+  req: MedusaRequest,
+  res: MedusaResponse
+) => {
+  const query = req.scope.resolve(ContainerRegistrationKeys.QUERY)
+
+  const { data: myCustoms } = await query.graph({
+    entryPoint: "my_custom",
+    fields: ["id", "name"],
+  })
+
+  res.json({ my_customs: myCustoms })
+}
+```
+
+In the above example, you resolve the Query from the Medusa container using the `ContainerRegistrationKeys.QUERY` (`query`) key.
+
+Then, you run a query using its `graph` method. This method accepts as a parameter an object with the following required properties:
+
+- `entryPoint`: The data model's name, as specified in the first parameter of the `model.define` method used for the data model's definition.
+- `fields`: An array of the data model’s properties to retrieve in the result.
+
+The method returns an object that has a `data` property, which holds an array of the retrieved data. For example:
+
+```json title="Returned Data"
+{
+  "data": [
+    {
+      "id": "123",
+      "name": "test"
+    }
+  ]
+}
+```
+
+---
+
+## Retrieve Linked Records
+
+Retrieve the records of a linked data model by passing in `fields` the data model's name suffixed with `.*`.
+
+For example:
+
+```ts highlights={[["6"]]}
+const { data: myCustoms } = await query.graph({
+  entryPoint: "my_custom",
+  fields: [
+    "id", 
+    "name",
+    "product.*",
+  ],
+})
+```
+
+<Note title="Tip">
+
+`.*` means that all of data model's properties should be retrieved. To retrieve a specific property, replace the `*` with the property's name. For example, `product.title`.
+
+</Note>
+
+### Retrieve List Link Records
+
+If the linked data model has `isList` enabled in the link definition, pass in `fields` the data model's plural name suffixed with `.*`.
+
+For example:
+
+```ts highlights={[["6"]]}
+const { data: myCustoms } = await query.graph({
+  entryPoint: "my_custom",
+  fields: [
+    "id", 
+    "name",
+    "products.*",
+  ],
+})
+```
+
+---
+
+## Apply Filters
```

Comment:
question: Should we make it clear that the `graph` engine does not support cross-module filtering?

Reply:
Same goes for sorting.

Reply:
Makes sense, will add 👍🏻